src/pybind/mgr/cephadm/upgrade.py

```diff
@@ -17,6 +17,20 @@
 logger = logging.getLogger(__name__)
 
 
+def normalize_image_digest(digest: str, default_registry: str) -> str:
+    # normal case:
+    #   ceph/ceph -> docker.io/ceph/ceph
+    # edge cases that shouldn't ever come up:
+    #   ubuntu -> docker.io/ubuntu    (ubuntu alias for library/ubuntu)
+    # no change:
+    #   quay.ceph.io/ceph/ceph -> ceph
+    #   docker.io/ubuntu -> no change
+    bits = digest.split('/')
+    if '.' not in bits[0] or len(bits) < 3:
+        digest = 'docker.io/' + digest
```

Comment:
unfortunately, it was needed also to use the constant "DEFAULT_REGISTRY" in cephadm orchestrator...  i will put another pr to avoid manually replace the defaukt registry in downstream distrinbutions